ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueConsumption.scala

```diff
@@ -261,6 +295,23 @@ object KeyValueConsumption {
     val hexTxId = parseLedgerString("TransactionId")(
       BaseEncoding.base16.encode(entryId.toByteArray)
     )
+
+    val maybeBlindingInfoWithDivulgedContracts =
+      if (txEntry.hasBlindingInfo)
+        Some(Conversions.decodeBlindingInfo(txEntry.getBlindingInfo))
+      else
+        None
+
+    maybeBlindingInfoWithDivulgedContracts.foreach { case (_, divulgedContracts) =>
+      checkDivulgence(divulgedContracts, hexTxId)
+    }
+
+    val divulgedContracts = maybeBlindingInfoWithDivulgedContracts
+      .map(_._2.iterator.collect { case (contractId, Some(contractInstance)) =>
+        DivulgedContract(contractId, contractInstance)
+      }.toList)
+      .getOrElse(List.empty[DivulgedContract])
```

Comment:
I think extracting `maybeBlindingInfoWithDivulgedContracts` via pattern matching would be more readable (mostly due to the pair, you could also consider a case class).

Reply:
Done